csharp/ql/lib/ext/System.model.yml

```diff
@@ -374,12 +374,37 @@ extensions:
       - ["System", "Lazy<T>", False, "Lazy", "(System.Func<T>,System.Boolean)", "", "Argument[0].ReturnValue", "Argument[this].Property[System.Lazy`1.Value]", "value", "manual"]
       - ["System", "Lazy<T>", False, "Lazy", "(System.Func<T>,System.Threading.LazyThreadSafetyMode)", "", "Argument[0].ReturnValue", "Argument[this].Property[System.Lazy`1.Value]", "value", "manual"]
       - ["System", "Lazy<T>", False, "get_Value", "()", "", "Argument[this]", "ReturnValue", "taint", "manual"]
+      - ["System", "MemoryExtensions", False, "Replace<T>", "(System.Span<T>,T,T)", "", "Argument[2]", "Argument[0].Element", "value", "manual"]
+      - ["System", "MemoryExtensions", False, "Replace<T>", "(System.ReadOnlySpan<T>,System.Span<T>,T,T)", "", "Argument[0].Element", "Argument[1].Element", "value", "manual"]
+      - ["System", "MemoryExtensions", False, "Replace<T>", "(System.ReadOnlySpan<T>,System.Span<T>,T,T)", "", "Argument[3]", "Argument[1].Element", "value", "manual"]
       - ["System", "Nullable<T>", False, "GetValueOrDefault", "()", "", "Argument[this].Property[System.Nullable`1.Value]", "ReturnValue", "value", "manual"]
       - ["System", "Nullable<T>", False, "GetValueOrDefault", "(T)", "", "Argument[0]", "ReturnValue", "value", "manual"]
       - ["System", "Nullable<T>", False, "GetValueOrDefault", "(T)", "", "Argument[this].Property[System.Nullable`1.Value]", "ReturnValue", "value", "manual"]
       - ["System", "Nullable<T>", False, "Nullable", "(T)", "", "Argument[0]", "Argument[this].Property[System.Nullable`1.Value]", "value", "manual"]
       - ["System", "Nullable<T>", False, "get_HasValue", "()", "", "Argument[this].Property[System.Nullable`1.Value]", "ReturnValue", "taint", "manual"]
       - ["System", "Nullable<T>", False, "get_Value", "()", "", "Argument[this]", "ReturnValue", "taint", "manual"]
+      - ["System", "ReadOnlySpan<T>", False, "CopyTo", "(System.Span<T>)", "", "Argument[this].Element", "Argument[0].Element", "value", "manual"]
+      - ["System", "ReadOnlySpan<T>", False, "get_Item", "(System.Int32)", "", "Argument[this].Element", "ReturnValue", "value", "manual"]
+      - ["System", "ReadOnlySpan<T>", False, "GetPinnableReference", "()", "", "Argument[this].Element", "ReturnValue", "value", "manual"]
```

Comment:
Does a call to `GetPinnableReference` gets extracted when we face a `fixed (int* spanPtr = span) {}` statement?

Reply:
Not as far as I can tell. Compiling and decompiling the code yields a call to `GetPinnableReference`, but we are not extracting it.
Have you run into something similar before and/or what is the reason for asking?

Reply:
I was just asking because the docs on `GetPinnableReference` says that it shouldn't be called directly. So I think we'd only get value out of this row, if it was extracted.

Reply:
Yes, that is probably true. In any case, I think we should keep it in as it is a part of the public API.